delta2cons/delta2cons.go

```diff
@@ -0,0 +1,393 @@
+			err := d2c.processOngoingTick(ctx)
+			if err != nil {
+				return err
+			}
+		}
+	}
+}
+
+func (d2c *Delta2Cons) handleDeltaManifest(ctx context.Context, dm *logstream.DeltaManifest) error {
+	for commandID, cmd := range dm.GetFields().GetCommands() {
+		cm, ok := d2c.manifest.GetCommands()[commandID]
+		if !ok {
+			return fmt.Errorf("command %q not found in manifest", commandID)
+		}
+		var tm *logstream.TargetManifest
+		if cm.GetTargetId() != "" {
+			var ok bool
+			tm, ok = d2c.manifest.GetTargets()[cm.GetTargetId()]
+			if !ok {
+				return fmt.Errorf("target %s not found in manifest", cm.GetTargetId())
+			}
+		}
+		if cmd.GetStatus() == logstream.RunStatus_RUN_STATUS_IN_PROGRESS {
+			d2c.printHeader(d2c.console, cm.GetTargetId(), commandID, tm, cm)
+		}
+		if cmd.GetHasHasProgress() && d2c.shouldPrintProgress(cm.GetTargetId(), commandID, cm) {
+			d2c.printProgress(cm.GetTargetId(), commandID, cm)
+		}
+		if cmd.GetStatus() == logstream.RunStatus_RUN_STATUS_FAILURE {
+			d2c.printError(cm.GetTargetId(), commandID, tm, cm)
+		}
+	}
+	if dm.GetFields().GetHasFailure() {
+		d2c.printBuildFailure()
+	}
+	return nil
+}
+
+func (d2c *Delta2Cons) getCommand(commandID string) *command {
+	cmd, ok := d2c.commands[commandID]
+	if !ok {
+		cmd = &command{}
+		d2c.commands[commandID] = cmd
+	}
+	return cmd
+}
+
+func (d2c *Delta2Cons) handleDeltaLog(ctx context.Context, dl *logstream.DeltaLog) error {
+	c := d2c.console
+	if dl.GetTargetId() == "" && strings.HasPrefix("_generic:", dl.GetCommandId()) {
+		prefix := strings.TrimPrefix(dl.GetCommandId(), "_generic:")
+		c = c.WithPrefixAndSalt(prefix, prefix)
+	} else if dl.GetTargetId() != "" {
+		tm, ok := d2c.manifest.GetTargets()[dl.GetTargetId()]
+		if !ok {
+			return fmt.Errorf("target %s not found in manifest", dl.GetTargetId())
+		}
+		c = c.WithPrefixAndSalt(tm.GetName(), dl.GetTargetId())
+	} else {
+		c = c.WithPrefixAndSalt(dl.GetCommandId(), dl.GetCommandId())
+	}
+	cmd := d2c.getCommand(dl.GetCommandId())
+
+	sameAsLast := (!d2c.lastOutputWasOngoingUpdate &&
+		!d2c.lastOutputWasProgress &&
+		d2c.lastCommandOutput == cmd)
+	output := dl.GetData()
+	printOutput := make([]byte, 0, len(cmd.openLine)+len(output)+10)
+	if bytes.HasPrefix(output, []byte{'\n'}) && len(cmd.openLine) > 0 && !cmd.lastOpenLineSkipped {
+		// Optimization for cases where ansi control sequences are not supported:
+		// if the output starts with a \n, then treat the open line as closed and
+		// just keep going after that.
+		cmd.openLine = nil
+		output = output[1:]
+		cmd.lastOpenLineUpdate = time.Time{}
+	}
+	if sameAsLast && len(cmd.openLine) > 0 {
+		// Prettiness optimization: if there is an open line and the previous print out
+		// was of the same vertex, then use ANSI control sequence to go up one line and
+		// keep writing there.
+		printOutput = append(printOutput, ansiUp...)
+	}
+	// Prepend the open line to the output.
+	printOutput = append(printOutput, cmd.openLine...)
```

Comment:
Is this meant to be 
```suggestion
	printOutput = append(cmd.openLine, printOutput...)
```
?

Reply:
No... I'm using the builder pattern here, where `printOutput` is getting built piece by piece along the way. The line following this actually adds the `output` after the open line.